doctr/models/modules/vision_transformer/pytorch.py

```diff
@@ -24,14 +25,53 @@ def __init__(
 
         super().__init__()
         channels, height, width = input_shape
-        self.patch_size = patch_size
+        # fix patch size if recognition task with 32x128 input
+        self.patch_size = (4, 8) if height != width else patch_size
```

Comment:
tricky condition: what are all the possible cases as input, and what do we want as patch_size for each?

Reply:
oh it could be anything ..
currently in classification case: 32x32 (check)
recognition case: 32x128 (check)
detection case 1024x1024 (not handled)
any other size (not handled) 

it will not fail but each size needs a different patch_size

Reply:
Ok two questions then:
- how should the scale impact the patch size ? (N,N) --> (H,W) implies that (2N,2N) --> (?,?)
- how should the aspect ratio impact the patch size? I see that (32,32) --> (4, 4), but why (32,128) doesn't do (4, 16) ?

Reply:
https://github.com/baudm/parseq/blob/main/configs/model/parseq.yaml
https://github.com/baudm/parseq/blob/main/configs/experiment/vitstr.yaml